crates/storage/src/state.rs

```diff
@@ -341,6 +341,44 @@ impl StarknetBlocksTable {
         .map_err(|e| e.into())
     }
 
+    /// Returns the state commitment of the given block.
+    pub fn get_state_commitment(
+        tx: &Transaction<'_>,
+        block: StarknetBlocksBlockId,
+    ) -> anyhow::Result<Option<(StorageCommitment, ClassCommitment)>> {
+        let mut statement = match block {
+            StarknetBlocksBlockId::Number(_) => {
+                tx.prepare("SELECT root, class_commitment FROM starknet_blocks WHERE number = ?")
+            }
+            StarknetBlocksBlockId::Hash(_) => {
+                tx.prepare("SELECT root, class_commitment FROM starknet_blocks WHERE hash = ?")
+            }
+            StarknetBlocksBlockId::Latest => tx.prepare(
+                "SELECT root, class_commitment FROM starknet_blocks ORDER BY number DESC LIMIT 1",
+            ),
+        }?;
+
+        let mut rows = match block {
+            StarknetBlocksBlockId::Number(number) => statement.query([number]),
+            StarknetBlocksBlockId::Hash(hash) => statement.query([hash]),
+            StarknetBlocksBlockId::Latest => statement.query([]),
+        }?;
+
+        let row = rows.next().context("Iterate rows")?;
+
+        match row {
+            Some(row) => {
+                let storage_commitment = row.get_unwrap("root");
+                let class_commitment = row
+                    .get_unwrap::<_, Option<_>>("class_commitment")
+                    .unwrap_or(ClassCommitment::ZERO);
+
+                Ok(Some((storage_commitment, class_commitment)))
+            }
+            None => Ok(None),
+        }
+    }
```

Comment:
Hmm okay so we never store the `StateCommitment` but rather calculate it on the fly if we need it. Did you consider storing it as another column? This function would then read both it and `root`, and then if `state_commitment` is null then it defaults to `root` for backwards compat?

Reply:
Yeah, right now it's calculated on demand and not stored anywhere, the column would be somewhat redundant but at the same time we wouldn't have to recalculate the hash. I'm not sure which one is better right now.

Reply:
Personally I wouldn't bother storing the state commitment for now. Calculating the Pedersen hash is ~30 µs on my hardware, so it shouldn't have too big an impact for JSON-RPC / L2 sync right now. (Handling a single JSON-RPC `starknet_getBlockWithTxHashes` is ~5.5 ms.)